test/unit/compose/volume.spec.ts

```diff
@@ -314,4 +318,19 @@ describe('compose/volume: unit tests', () => {
 			).to.be.true;
 		});
 	});
+
+	describe('extra firmware volume handling', () => {
+		it('should throw ExtraFirmwareVolumeError when parsing extra-firmware volume from Docker', () => {
+			expect(() =>
+				Volume.fromDockerVolume({
+					Driver: 'local',
+					Name: EXTRA_FIRMWARE_VOLUME_NAME,
+					Mountpoint: `/var/lib/docker/volumes/${EXTRA_FIRMWARE_VOLUME_NAME}/_data`,
```

Comment:
@acostach @cywang117, by reading your code I'm not too clear on what's going to the config.json.
On the OS side we are waiting for `<volume_name>/_data`, and I suspect that you are only giving us the `<volume_name>`, personally I'd rather have the full path so that the engine dependency is completely on the supervisor side, and we basically only work with directories, and no clever stuff related to Docker.
See: https://github.com/balena-os/meta-balena/pull/3754#discussion_r2585224724

Reply:
We discussed this internally and came to the conclusion that the OS scripts will be updated to infer `/var/lib/docker/volumes/$VOLUME_NAME/_data` from `$VOLUME_NAME`, as that path could change in the future and having the Supervisor write it to config.json in full would necessitate a reboot from the user to get to firmware at the updated path.